ra/ra.go

```diff
@@ -763,43 +763,33 @@ func (ra *RegistrationAuthorityImpl) checkAuthorizationsCAA(
 func (ra *RegistrationAuthorityImpl) recheckCAA(ctx context.Context, names []string) error {
 	ra.stats.Inc("recheck_caa", 1)
 	ra.stats.Inc("recheck_caa_names", int64(len(names)))
-	wg := sync.WaitGroup{}
-	ch := make(chan *probs.ProblemDetails, len(names))
+	ch := make(chan *probs.ProblemDetails)
```

Comment:
While we are here, we never actually use anything but the `Detail` field of the `ProblemDetails` we construct/pass through. Can you change this to a `chan string`, then we don't have to construct dummy `ProblemDetails` just to carry the detail.

Reply:
I had contemplated doing that, however there are a couple of gotchas - if you use chan string then the only way to indicate success is via an empty string. If you ever end up with resp.Problem.Detail being non-populated (especially if you move to proto3, where it drops string pointers), then you'll end up with a "success" when you actually want a "failure". One alternative is to use chan *string, however that has a similar but different set of problems.

The other thing is that I believe there is another bug in this code (per issue https://github.com/letsencrypt/boulder/issues/3143), where the ServerInternal failure gets blindly mapped to a CAAError. I suspect the appropriate fix may be to actually use a chan error and distinguish between error types. Using a chan string now just complicates this (or at least has to be partially unwound).

Reply:
Yeah, I'm a little happier keeping the type of the chan as `*probs.ProblemDetails`.